standard/expressions.md

```diff
@@ -541,7 +541,7 @@ argument_name
 argument_value
     : expression
     | 'ref' variable_reference
-    | 'out' variable_reference
+    | 'out' local_variable_type? identifier
```

Comment:
The speclets for out vars and deconstructions relied on `declaration_expression`.
But in this doc, I see `'out' local_variable_type? identifier` and in the deconstruction [draft](https://github.com/dotnet/csharpstandard/pull/63/files#diff-d9d678d0b42d3c668ebd64159dadfd6188361121feca8fb94fddf8816433cd14R6357) I see `tuple_deconstruction_expression`.
I'm not sure what is the benefit of this structure.

I think having a `declaration_expression` that appears in multiple contexts is a good factorization. Then we don't need new syntax for deconstructions (it's `declaration_expression` and `tuple_expression`).

Tagging @cston since I'll be away.

Reply:
Some more context on syntax shape (was discussed with LDM): https://github.com/dotnet/csharpstandard/issues/285

Reply:
Updated to use `declaration_expression`.

Tagging @Nigel-Ecma for grammar checks.

Reply:
*declaration_expression* is different and cannot simply be substituted in this location. Here *local_variable_type* is optional, it is not in *declaration_expression*. I'll find the change and add a suggestion there…